frontend-test/unit/schemaServiceSpec.js

```diff
@@ -618,5 +618,114 @@ define([
         expect(result).toBeFalsy();
       });
     });
+
+    describe('extractPvfs', () => {
+      it('should return pvfs keyed by criterion id', () => {
+        const criteria = {
+          crit1Id: {
+            id: 'crit1Id',
+            title: 'criterion1',
+            dataSources: [
+              {id: 'ds1Id', pvf: {range: [0, 1], direction: 'increasing'}}
+            ]
+          }
+        };
+        const result = schemaService.extractPvfs(criteria);
+        const expectedResult = {
+          crit1Id: {
+            direction: 'increasing'
+          }
+        };
+        expect(result).toEqual(expectedResult);
+      });
+    });
+
+    describe('mergePvfs', () => {
```

Comment:
Should we add a test that tests the case where one criterion is set on the scenario and another is not? Or would that be redundant with the existing tests?

Reply:
we test it with the second case